.github/workflows/node.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2024 Hemi Labs, Inc.
+# Use of this source code is governed by the MIT License,
+# which can be found in the LICENSE file.
+
+# GitHub Actions workflow to lint, build and test NPM packages.
+name: "Node"
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: "npm-${{ github.workflow }}-${{ github.event.number || github.ref }}"
+  cancel-in-progress: "${{ github.event_name == 'pull_request' }}"
+
+env:
+  GO_VERSION: "1.22.x"
+  PNPM_VERSION: "9.4.x"
+  NODE_VERSION: ">=20.0.0"
+  BINARYEN_VERSION: "117"
+
+jobs:
+  build:
+    name: "Build"
+    runs-on: "ubuntu-latest"
+    permissions:
+      contents: read
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+
+      - name: "Setup Go ${{ env.GO_VERSION }}"
+        uses: actions/setup-go@v5
+        with:
+          go-version: "${{ env.GO_VERSION }}"
+          cache: true
+          check-latest: true
+
+      - name: "Setup pnpm ${{ env.PNPM_VERSION }}"
+        uses: pnpm/action-setup@v4
+        with:
+          version: "${{ env.PNPM_VERSION }}"
+
+      - name: "Setup Node ${{ env.NODE_VERSION }}"
+        uses: actions/setup-node@v4
+        with:
+          node-version: "${{ env.NODE_VERSION }}"
```

Comment:
Not a blocker, but a minor suggestion here: You could commit a  `.nvmrc` file with the node version, so all contributors of this package know which version is needed. `actions/setup-node@v4` has support for `.nvmrc` files (and if using `nvm` locally you also to automatically switch to the correct node version when using this package locally 🎉 )

Reply:
Ah, will do, thanks!